src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpFindReferences.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpFindReferences : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpFindReferences(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpFindReferences))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.FindReferences)]
+        public void FindReferencesToCtor()
+        {
+            SetUpEditor(@"
+class Program
+{
+}$$
+");
+
+            VisualStudio.Instance.SolutionExplorer.AddFile(ProjectName, "File2.cs");
+            VisualStudio.Instance.SolutionExplorer.OpenFile(ProjectName, "File2.cs");
+
+            SetUpEditor(@"
+class SomeOtherClass
+{
+    void M()
+    {
+        Program p = new Progr$$am();
+    }
+}
+");
+
+            SendKeys(Shift(VirtualKey.F12));
+
+            VisualStudioWorkspaceOutOfProc.WaitForAsyncOperations(FeatureAttribute.FindReferences);
```

Comment:
My worry with this pattern is tests then have to think about where to put waiters. Historically, this has been one of the biggest sources of flakiness in our integration tests. Can the wait be placed somewhere else?